### PR TITLE
Validator slasher grpc client

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**

In order to enable slashing protection validators may choose to validate that their attestation and blocks are not slashable by connecting to a running slasher instance

**Which issues(s) does this PR fix?**

Part of #5806

### CODE REVIEW

**Comment by shayzluf** [comment]
@terencechain thanks for the feedback:
will work on the design doc today
about 1) it can also enable redundant validator setup using a slasher endpoint and firstly better protect validators for slashing
Screen Shot 2020-05-11 at 9 08 55

**Comment by shayzluf** [comment]
@terencechain added short design and flow https://hackmd.prylabs.network/-jrBcMw3Rz60Q-J9MCHpuQ?view

**Comment by shayzluf** [comment]
@rauljordan that design seems sound. staring to do it now

File: shared/logutil/logutil.go [hunk 1]
```diff
@@ -23,3 +27,24 @@ func ConfigurePersistentLogging(logFileName string) error {
 	logrus.Info("File logging initialized")
 	return nil
 }
+
+// LogDebugRequestInfoUnaryInterceptor this method logs the gRPC backend as well as request duration when the log level is set to debug
+// or higher.
+func LogDebugRequestInfoUnaryInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply shared/logutil/logutil.go:33]: I will make a grpc util?

**Comment by prestonvanloon** [comment]
This looks very similar to #5895. What is going on with these?

**Comment by shayzluf** [comment]
> This looks very similar to #5895. What is going on with these?
5895 builds off this one

Screen Shot 2020-05-19 at 6 56 00

File: shared/logutil/logutil.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nisdas) [reply shared/logutil/logutil.go:33]: Yes maybe that would be better

[the remark this answers is not in the document]
Comment (shayzluf) [reply shared/logutil/logutil.go:33]: closing this pr.
will implement the changed on #5895

**Pull request closed without merging**